src/DGmethods/NumericalFluxes.jl

```diff
@@ -1,82 +1,147 @@
+                     bctype, t)
+  numerical_flux!(dnf, bl, F, nM, QM, QVM, auxM, QP, QVP, auxP, t)
+end
+
+
 
 """
     rusanov!(F::MArray, nM, QM, QVM, auxM, QP, QVP, auxP, t, flux!, wavespeed,
-             [computeQjump!])
+             [preflux = (_...) -> (), computeQjump!])
 
 Calculate the Rusanov (aka local Lax-Friedrichs) numerical flux given the plus
 and minus side states/viscous states `QP`/`QVP` and `QM`/`QVM` using the physical
 flux function `flux!` and `wavespeed` calculation.
 
 The `flux!` has almost the same calling convention as `flux!` from
-[`DGBalanceLaw`](@ref).
+[`DGBalanceLaw`](@ref) except that `preflux(Q, aux, t)` is splatted at the end
+of the call.
 
 The function `wavespeed` should return the maximum wavespeed for a state and is
-called as `wavespeed(nM, QM, auxM, t)` and `wavespeed(nM, QP, auxP, t)` where
-`nM` is the outward unit normal for the minus side.
+called as `wavespeed(nM, QM, auxM, t, preflux(QM, auxM, t)...)` and
+`wavespeed(nM, QP, auxP, t, preflux(QP, auxP, t)...)` where `nM` is the outward
+unit normal for the minus side.
 
 When present `computeQjump!(ΔQ, QM, auxM, QP, auxP)` will be called after so
 that the user specify the value to use for `QM - QP`; this is useful for
 correcting `Q` to include discontinuous reference states.
 
+!!! note
+
+    The undocumented arguments `PM` and `PP` for the function should not be used
+    by external callers and are used only internally by the function
+    `rusanov_boundary_flux!`
+
 """
-function rusanov!(F::MArray{Tuple{nstate}}, nM,
-                  QM, QVM, auxM,
-                  QP, QVP, auxP,
-                  t, flux!, wavespeed,
-                  computeQjump! = nothing
-                 ) where {nstate}
-  λM = wavespeed(nM, QM, auxM, t)
-  FM = similar(F, Size(3, nstate))
-  flux!(FM, QM, QVM, auxM, t)
+struct Rusanov <: DivNumericalFlux
+end
+
+
+function numerical_flux!(::Rusanov, bl::BalanceLaw,
+                         F::MArray, nM,
+                         QM, QVM, auxM,
+                         QP, QVP, auxP,
+                         t)
+  DFloat = eltype(F)
+  nstate = num_state(bl,DFloat)
 
-  λP = wavespeed(nM, QP, auxP, t)
+  λM = wavespeed(bl, nM, Vars{vars_state(bl,DFloat)}(QM), Vars{vars_aux(bl,DFloat)}(auxM), t)
+  FM = similar(F, Size(3, nstate))
+  flux!(bl, Grad{vars_state(bl,DFloat)}(FM), Vars{vars_state(bl,DFloat)}(QM), Vars{vars_diffusive(bl,DFloat)}(QVM), Vars{vars_aux(bl,DFloat)}(auxM), t)
+  
+  λP = wavespeed(bl, nM, Vars{vars_state(bl,DFloat)}(QP), Vars{vars_aux(bl,DFloat)}(auxP), t)
   FP = similar(F, Size(3, nstate))
-  flux!(FP, QP, QVP, auxP, t)
+  flux!(bl, Grad{vars_state(bl,DFloat)}(FP), Vars{vars_state(bl,DFloat)}(QP), Vars{vars_diffusive(bl,DFloat)}(QVP), Vars{vars_aux(bl,DFloat)}(auxP), t)
 
   λ  =  max(λM, λP)
 
-  if computeQjump! === nothing
+  # if computeQjump! === nothing
```

Comment:
Since this is commented out I'm not sure what `isentropic_vortex_standalone_aux` is passing. That's a bit worrying...

Reply:
Oh I see now that `src/DGmethods/DGBalanceLawDiscretizations.jl` now includes `include("NumericalFluxes_old.jl")`